backend/src/migrations/20240221213010_add_user_achievements_table.ts

```diff
@@ -0,0 +1,35 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'user_achievements';
+
+const ColumnName = {
+    USER_ID: 'user_id',
+    ACHIEVEMENT_ID: 'achievement_id',
+    DATE_ACHIEVED: 'date_achieved',
+};
+
+async function up(knex: Knex): Promise<void> {
+    return knex.schema.createTable(TABLE_NAME, (table) => {
+        table.increments().primary();
+        table.integer(ColumnName.USER_ID).unsigned().notNullable();
+        table
+            .foreign(ColumnName.USER_ID)
+            .references('users.id')
```

Comment:
Let's switch to using references().inTable() instead, this notation is a bit clearer. Also, let's include the foreign key directly in the column creation, as we did in previous migrations.